forge-ai/src/main/java/forge/ai/ComputerUtilMana.java

```diff
@@ -666,47 +666,42 @@ private static boolean payManaCost(final ManaCostBeingPaid cost, final SpellAbil
 
         List<Mana> manaSpentToPay = test ? new ArrayList<>() : sa.getPayingMana();
         List<SpellAbility> paymentList = Lists.newArrayList();
+        final ManaPool manapool = ai.getManaPool();
 
-        if (ManaPool.payManaCostFromPool(cost, sa, ai, test, manaSpentToPay)) {
-            return true;    // paid all from floating mana
-        }
-
-        boolean purePhyrexian = cost.containsOnlyPhyrexianMana();
-
+        // Apply the color/type conversion matrix if necessary
+        manapool.restoreColorReplacements();
         boolean ignoreColor = false, ignoreType = false;
-        StaticAbility mayPlay = sa.getMayPlay();
+        CardPlayOption mayPlay = sa.getHostCard().mayPlay(sa.getMayPlay());
         if (mayPlay != null) {
-            if (mayPlay.hasParam("MayPlayIgnoreColor")) {
-                ignoreColor = true;
-            } else if (mayPlay.hasParam("MayPlayIgnoreType")) {
+            if (mayPlay.isIgnoreManaCostType()) {
                 ignoreType = true;
+            } else if (mayPlay.isIgnoreManaCostColor()) {
+                ignoreColor = true;
             }
-        } else if (sa.hasParam("ActivateIgnoreColor")) {
+            mayPlay.applyManaConvert(manapool);
+        } else if (sa.isActivatedAbility() && sa.getGrantorStatic() != null && sa.getGrantorStatic().hasParam("ManaConversion")) {
+            AbilityUtils.applyManaColorConversion(manapool, sa.getGrantorStatic().getParam("ManaConversion"));
             ignoreColor = true;
         }
+        StaticAbilityManaConvert.manaConvert(manapool, ai, sa.getHostCard(), sa);
+
+        if (ManaPool.payManaCostFromPool(cost, sa, ai, test, manaSpentToPay)) {
+            return true;    // paid all from floating mana
+        }
+
+        boolean purePhyrexian = cost.containsOnlyPhyrexianMana();
         boolean hasConverge = sa.getHostCard().hasConverge();
         ListMultimap<ManaCostShard, SpellAbility> sourcesForShards = getSourcesForShards(cost, sa, ai, test,
                 checkPlayable, hasConverge, ignoreColor, ignoreType);
 
         int testEnergyPool = ai.getCounters(CounterEnumType.ENERGY);
-        final ManaPool manapool = ai.getManaPool();
         ManaCostShard toPay = null;
         List<SpellAbility> saExcludeList = new ArrayList<>();
 
         // Loop over mana needed
         while (!cost.isPaid()) {
             while (!cost.isPaid() && !manapool.isEmpty()) {
                 boolean found = false;
-
-                // Apply the color/type conversion matrix if necessary
-                final CostPayment pay = new CostPayment(sa.getPayCosts(), sa);
```

Comment:
@tool4ever I don't know if the removal of CostPayment is okay?

Reply:
Well I thought if all that happens is creating a fresh Matrix only to copy it later onto the reset ManaPool it's not needed

Currently everything seems to work but this needs more tests later